packages/react-devtools-shared/src/devtools/views/Components/EditableValue.js

```diff
@@ -44,6 +45,15 @@ export default function EditableValue({
       externalValue: value,
     });
 
+  const handeCheckBoxToggle = ({target}) => {
+    dispatch({
+      type: 'UPDATE',
+      editableValue: target.checked,
+      externalValue: value,
+    });
+    applyChangesFromCheckBox();
+  };
```

Comment:
```suggestion
  const handeCheckBoxToggle = ({target}) => {
    dispatch({
      type: 'UPDATE',
      editableValue: target.checked,
      externalValue: value,
    });

    // Unlike <input type="text"> which has both an onChange and an onBlur,
    // <input type="checkbox"> updates state *and* applies changes in a single event.
    // So we read from target.checked rather than parsedValue (which has not yet updated).
    // We also don't check isValid (because that hasn't changed yet either);
    // we don't need to check it anyway, since target.checked is always a boolean.
    overrideValueFn(path, target.checked);
  };
```

Reply:
got it! this makes sense